certinel_test.go

```diff
@@ -234,3 +236,43 @@ func TestClose(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func TestGetClientCertificate(t *testing.T) {
+	// Set up a TLS server that will ask for client certs.
+	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	}))
+	ts.TLS = &tls.Config{
+		ClientAuth: tls.RequestClientCert,
```

Comment:
Should we set the value to `tls.RequireAnyClientCert` to verify we're not sending any certificates and the connection is rest?